gobblin-data-management/src/main/java/gobblin/data/management/dataset/DatasetUtils.java

```diff
@@ -63,7 +63,7 @@ public boolean accept(Path path) {
    * @throws IOException
    */
   @SuppressWarnings("unchecked")
-  public static <T extends Dataset> DatasetFinder<T> instantiateDatasetFinder(Properties props, FileSystem fs,
+  public static <T extends gobblin.dataset.Dataset> DatasetFinder<T> instantiateDatasetFinder(Properties props, FileSystem fs,
```

Comment:
That is an interesting point, and we should probably do that in the future. However, I would want to push that to a different PR.